src/coreComponents/constitutive/contact/Coulomb.hpp

```diff
@@ -27,11 +28,63 @@ namespace constitutive
 {
 
 /**
- * @class MohrCoulomb
+ * @class CoulombUpdates
+ *
+ * This class is used for in-kernel contact relation updates
+ */
+class CoulombUpdates : public ContactBaseUpdates
+{
+public:
+
+  CoulombUpdates( real64 const & cohesion,
+                  real64 const & frictionCoefficient )
+    : m_cohesion( cohesion ),
+    m_frictionCoefficient( frictionCoefficient )
+  {}
+
+  /// Default copy constructor
+  CoulombUpdates( CoulombUpdates const & ) = default;
+
+  /// Default move constructor
+  CoulombUpdates( CoulombUpdates && ) = default;
+
+  /// Deleted default constructor
+  CoulombUpdates() = delete;
+
+  /// Deleted copy assignment operator
+  CoulombUpdates & operator=( CoulombUpdates const & ) = delete;
+
+  /// Deleted move assignment operator
+  CoulombUpdates & operator=( CoulombUpdates && ) =  delete;
+
+  /**
+   * @brief Evaluate the limit tangential traction norm and return the derivative wrt normal traction
+   * @param[in] normalTraction the normal traction
+   * @param[out] dLimitTangentialTractionNorm_dTraction the derivative of the limit tangential traction norm wrt normal traction
+   * @return the limit tangential traction norm
+   */
+  GEOSX_HOST_DEVICE
+  inline
+  virtual real64 computeLimitTangentialTractionNorm( real64 const & normalTraction,
+                                                     real64 & dLimitTangentialTractionNorm_dTraction ) const override;
+
+private:
+
+  /// The cohesion for each upper level dimension (i.e. cell) of *this
+  real64 m_cohesion;
+
+  /// The friction coefficient for each upper level dimension (i.e. cell) of *this
+  real64 m_frictionCoefficient;
+
+};
+
+
+/**
+ * @class Coulomb
  *
  * Class to provide a Coulomb friction model.
  */
-class Coulomb : public ContactRelationBase
+class Coulomb : public ContactBase
```

Comment:
maybe `CoulombContact` or `CoulombFriction` would be a better name?

Reply:
Done, now the class is called `CoulombContact`.